lib/npm.js

```diff
@@ -32,8 +32,13 @@ var NPMLocation = function(options, ui) {
   this.name = options.name;
 
   var npmrc = new Npmrc();
-  
-  this.registryURL = options.registry || npmrc.getRegistry() || defaultRegistry;
+
+  this.registryURL = function (scope) {
+    if (scope) {
+      return npmrc.getRegistry(scope) || options.registry || defaultRegistry;
+    }
+    return options.registry || npmrc.getRegistry() || defaultRegistry;
```

Comment:
Perhaps we can just make this:

```
  return scope && npmrc.getRegistry(scope) || options.registry || npmrc.getRegistry() || defaultRegistry;
```

Note the fallback to `npmrc.getRegistry()` for scopes.